lib/urllib_kerberos/__init__.py

```diff
@@ -94,35 +88,37 @@ def generate_request_header(self, req, headers, neg_value):
             kwargs['principal'] = user
             kwargs['password'] = password
 
-        result, self.context = k.authGSSClientInit("HTTP@%s" % domain,
-                                                   **kwargs)
+        kwargs['mech_oid'] = self.mech_oid
+
+        result, req.kerberos_context = k.authGSSClientInit("HTTP@%s" % domain,
```

Comment:
Yes. For example, HTTPRedirectHandler:
        # loop detection
        # .redirect_dict has a key url if url was previously visited.
        if hasattr(req, 'redirect_dict'):
            visited = new.redirect_dict = req.redirect_dict
            if (visited.get(newurl, 0) >= self.max_repeats or
                len(visited) >= self.max_redirections):
                raise HTTPError(req.full_url, code,
                                self.inf_msg + msg, headers, fp)
        else:
            visited = new.redirect_dict = req.redirect_dict = {}